packages/yoroi-extension/chrome/extension/background.js

```diff
@@ -990,15 +990,26 @@ function handleInjectorConnect(port) {
                   },
                   connection
                 );
-              if (!returnTx && resp?.ok != null) {
-                const witnessSetResp = Buffer.from(
-                  RustModule.WalletV4.Transaction.from_bytes(
-                    Buffer.from(resp.ok, 'hex'),
-                  ).witness_set().to_bytes()
-                ).toString('hex');
-                rpcResponse({ ok: witnessSetResp });
-              } else {
+              if (resp?.ok == null) {
                 rpcResponse(resp);
+              } else if (returnTx && resp?.ok) {
+                const bodyOrTxBytes = Buffer.from(tx, 'hex');
+                let fullTx;
+                try {
+                  fullTx = RustModule.WalletV4.FixedTransaction.new(
+                    bodyOrTxBytes,
+                    resp.ok.to_bytes(),
+                    true,
+                  );
+                } catch {
+                  fullTx = RustModule.WalletV4.FixedTransaction.from_bytes(bodyOrTxBytes);
+                }
+                fullTx.set_witness_set(resp.ok);
```

Comment:
Be careful, `Fixed Transaction.set_witness_set` accepts serialized witness set

Reply:
good catch! Thanks @lisicky!